### PR TITLE
FA-87 Domain Unit Tests

### PR DESCRIPTION
**Pull request by brosscode**

### CODE REVIEW

File: createAccountCore/src/test/java/com/brandon/createaccount/ExampleUnitTest.kt [hunk 1]
```diff
@@ -1,17 +1,199 @@
 package com.brandon.createaccount
 
+import com.brandon.createaccount.core.data.mocks.CreateUserRepoAlwaysFailsMock
```
Comment (ForzenAcademy) [inline createAccountCore/src/test/java/com/brandon/createaccount/ExampleUnitTest.kt]: File name

File: loginCore/src/test/java/android/text/TextUtils.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package android.text;
+
+public class TextUtils {
+    public static boolean isDigitsOnly(CharSequence str) {
```
Comment (ForzenAcademy) [inline loginCore/src/test/java/android/text/TextUtils.java]: Copying java code is a no no. This can be done much more elegantly in Kotlin

Comment (brosscode) [reply loginCore/src/test/java/android/text/TextUtils.java]: Fixed by using a new extension function `String.isLoginCode()` in the extensions file. It is essentially `.isDigitsOnly()` by is a Kotlin version that can be used in unit tests without needing to be mocked.

File: createAccountCore/src/test/java/com/brandon/createaccount/ExampleUnitTest.kt (hunk 1 above)
Comment (brosscode) [reply createAccountCore/src/test/java/com/brandon/createaccount/ExampleUnitTest.kt]: Renamed

**Review by ForzenAcademy — APPROVED** [review]

**Pull request merged**